pkgs/misc/opcua-client-gui/default.nix

```diff
@@ -0,0 +1,33 @@
+{ lib
+, python3
+, fetchFromGitHub
+, python3Packages
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  version = "0.8.4";
+  pname = "opcua-client-gui";
+
+  src = fetchFromGitHub {
+    owner = "FreeOpcUa";
+    repo = pname;
+    rev = version;
+    hash = "sha256-0BH1Txr3z4a7iFcsfnovmBUreXMvIX2zpZa8QivQVx8=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    pyqt5
+    asyncua
+    opcua-widgets
+  ];
+
+  doCheck = false;
```

Comment:
same here

Reply:
In this case the tests don't work because they want the module [opcua](https://github.com/FreeOpcUa/python-opcua) which is deprecated by [asyncua](https://github.com/FreeOpcUa/opcua-asyncio) the software actually uses asyncio but the tests weren't updated yet.  
```
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/command/test.py:194: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated. Requirements should be satisfied by a PEP 517 installer. If you are using pip, you can try `pip install --use-pep517`.
  ir_d = dist.fetch_build_eggs(dist.install_requires)
/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/command/test.py:195: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated. Requirements should be satisfied by a PEP 517 installer. If you are using pip, you can try `pip install --use-pep517`.
  tr_d = dist.fetch_build_eggs(dist.tests_require or [])
/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/command/test.py:196: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated. Requirements should be satisfied by a PEP 517 installer. If you are using pip, you can try `pip install --use-pep517`.
  er_d = dist.fetch_build_eggs(
running egg_info
writing opcua_client.egg-info/PKG-INFO
writing dependency_links to opcua_client.egg-info/dependency_links.txt
writing entry points to opcua_client.egg-info/entry_points.txt
writing requirements to opcua_client.egg-info/requires.txt
writing top-level names to opcua_client.egg-info/top_level.txt
reading manifest file 'opcua_client.egg-info/SOURCES.txt'
adding license file 'LICENSE'
writing manifest file 'opcua_client.egg-info/SOURCES.txt'
running build_ext
SYS:PATH ['/build/source', '/build/source', '/nix/store/qf61iwnlhczqxckhk08awbs24c0ngimc-opcua-client-gui-0.8.4/lib/python3.10/site-packages', '/nix/store/95cxzy2hpizr23343b8bskl4yacf4b3l-python3-3.10.11/lib/python3.10/site-packages', '/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages', '/nix/store/ylz3xaryjhskfjbwc2p069zv9q2b3jj8-python3.10-wheel-0.38.4/lib/python3.10/site-packages', '/nix/store/160pf9z6zz4abgv34lxapvh6dixja9pm-python3.10-pip-23.0.1/lib/python3.10/site-packages', '/nix/store/70ajk510sal1l36kd5ph0ypdskj35qyn-python3.10-dbus-python-1.2.18/lib/python3.10/site-packages', '/nix/store/sbxgz372696lgr3phx83sbryh77jvws8-python3.10-pyqt5-sip-12.11.0/lib/python3.10/site-packages', '/nix/store/ai6jbsgcdg4zl9rqbr8s927lwmk3bpvp-python3.10-PyQt5-5.15.9/lib/python3.10/site-packages', '/nix/store/6hycdylb3hkaq7rizhrsq9c99figbvpm-python3.10-asyncua-1.0.2/lib/python3.10/site-packages', '/nix/store/gqwy59i9pk28vg5z40kdch5342f7bv7z-python3.10-aiosqlite-0.18.0/lib/python3.10/site-packages', '/nix/store/k7cvy7815nffj7pdpgsrjbxv9b3j1mp0-python3.10-aiofiles-23.1.0/lib/python3.10/site-packages', '/nix/store/y3f78pj8xa574mbl67v9l5q5543v1yjg-python3.10-pytz-2023.3/lib/python3.10/site-packages', '/nix/store/0v46g9dc22sq71alclfnqhz7v68sj48h-python3.10-python-dateutil-2.8.2/lib/python3.10/site-packages', '/nix/store/pbrzc0798laj1vndgjs0i3byjm3cgzcf-python3.10-six-1.16.0/lib/python3.10/site-packages', '/nix/store/zd88qz01wvipx24byi4vgflg0jwdqv1l-python3.10-sortedcontainers-2.4.0/lib/python3.10/site-packages', '/nix/store/6qjh115sbqzbjx3kii5i0wd2mpv0x75k-python3.10-cryptography-40.0.1/lib/python3.10/site-packages', '/nix/store/9r1dscsp2qfiixnyzah974lysgg9q5g9-python3.10-cffi-1.15.1/lib/python3.10/site-packages', '/nix/store/91g59ybxis8yahaarpciml2fd0rz7pwk-python3.10-pycparser-2.21/lib/python3.10/site-packages', '/nix/store/yd2chfzz0bcw0xb7da0g1iy6izfbks0r-python3.10-typing-extensions-4.5.0/lib/python3.10/site-packages', '/nix/store/yra6w6g0lchrjwwxj12cp4r97w32snvd-python3.10-opcua-widgets-0.6.1/lib/python3.10/site-packages', '/nix/store/95cxzy2hpizr23343b8bskl4yacf4b3l-python3-3.10.11/lib/python310.zip', '/nix/store/95cxzy2hpizr23343b8bskl4yacf4b3l-python3-3.10.11/lib/python3.10', '/nix/store/95cxzy2hpizr23343b8bskl4yacf4b3l-python3-3.10.11/lib/python3.10/lib-dynload']
PWD /build/source
pyqtgraph or numpy are not installed, use of graph feature disabled
tests (unittest.loader._FailedTest) ... ERROR

======================================================================
ERROR: tests (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests
Traceback (most recent call last):
  File "/nix/store/95cxzy2hpizr23343b8bskl4yacf4b3l-python3-3.10.11/lib/python3.10/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/nix/store/95cxzy2hpizr23343b8bskl4yacf4b3l-python3-3.10.11/lib/python3.10/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/build/source/tests.py", line 10, in <module>
    from opcua import ua
ModuleNotFoundError: No module named 'opcua'


----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (errors=1)
Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
error: Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
```

Reply:
Then also please add a short comment that the tests need to be updated

Reply:
Okay added it